index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+            {{GamepadEffectParameters/strongMagnitude}}, and
+            {{GamepadEffectParameters/weakMagnitude}}, none of which are
+            required because they default to 0.
+          </p>
+          <p>
+            {{GamepadEffectParameters/strongMagnitude}} and
+            {{GamepadEffectParameters/weakMagnitude}} set the vibration
+            intensity levels for the low-frequency and high-frequency rumble
+            motors, normalized to the range `[0,1]`, default 0. An effect
+            with magnitude values outside of this range is not a [=valid
+            effect=].
+          </p>
+          <p>
+            Given {{GamepadEffectParameters}} |params:GamepadEffectParameters|,
+            a <dfn>valid dual-rumble effect</dfn> must have a [=valid
+            duration=], a [=valid startDelay=], and both the
+            {{GamepadEffectParameters/strongMagnitude}} and the
+            {{GamepadEffectParameters/weakMagnitude}} must be in the range
+            `[0,1]`.
+          </p>
+        </dd>
+      </dl>
+    </section>
+    <section>
+      <h3>
+        <dfn>GamepadEffectParameters</dfn> Dictionary
+      </h3>
+      <p>
+        A <code>GamepadEffectParameters</code> dictionary contains keys for
+        parameters used by haptic effects. The meaning of each key is defined
+        by the haptic effect, and some keys may be unused.
+      </p>
+      <p>
+        To mitigate unwanted long-running effects, the [=user agent=] MAY
+        limit the total effect duration for a [=valid effect=] to some
+        maximum duration. It is RECOMMENDED that the [=user agent=] use a
+        maximum of 5 seconds.
+      </p>
+      <pre class="idl">
+        dictionary GamepadEffectParameters {
+            double duration = 0.0;
+            double startDelay = 0.0;
+            double strongMagnitude = 0.0;
+            double weakMagnitude = 0.0;
+        };
+      </pre>
+      <dl>
+        <dt>
+          <dfn>duration</dfn> member
+        </dt>
+        <dd>
+          {{GamepadEffectParameters/duration}} sets the duration of the
+          vibration effect in milliseconds. A <dfn>valid duration</dfn> must be
+          a positive value. An effect with an invalid
+          {{GamepadEffectParameters/duration}} is not a [=valid effect=]. The
+          default duration is 0 milliseconds.
+        </dd>
+        <dt>
+          <dfn>startDelay</dfn> member
+        </dt>
+        <dd>
+          {{GamepadEffectParameters/startDelay}} sets the duration of the
+          delay after {{GamepadHapticActuator/playEffect()}} is called
+          until vibration is started, in milliseconds. During the delay
+          interval, the actuator must not vibrate. A <dfn>valid startDelay</dfn>
+          must be a positive value. An effect with an invalid
+          {{GamepadEffectParameters/startDelay}} is not a [=valid effect=]. If
+          the {{GamepadEffectParameters/startDelay}} is not specified, the
```

Comment:
As above... just let WebIDL handler this through setting the data type to `unsigned long long`.

Reply:
I have changed them to `unsigned long long`, but I think that `unsigned short` would be sufficient for this purpose, which would allow the actuator to vibrate for at most 65535 milliseconds (or ~65 seconds) (https://webidl.spec.whatwg.org/#idl-unsigned-short).